src/target/renesas.c

```diff
@@ -558,9 +568,17 @@ static bool renesas_rv40_flash_write(target_flash_s *f, target_addr_t dest, cons
 
 	/* code flash or data flash operation */
 	const bool code_flash = dest < RENESAS_CF_END;
+	bool option_flash = false;
+
+	if (RENESAS_OF_START <= dest && dest <= (RENESAS_OF_START + RENESAS_OF_SIZE))
+		option_flash = true;
 
 	/* write size for code flash / data flash */
-	const uint8_t write_size = code_flash ? RV40_CF_WRITE_SIZE : RV40_DF_WRITE_SIZE;
+	uint8_t write_size = code_flash ? RV40_CF_WRITE_SIZE : RV40_DF_WRITE_SIZE;
+
+	/* Option-Setting write size is different to CF and DF memory */
+	if (option_flash)
+		write_size = RV40_OF_WRITE_SIZE;
```

Comment:
It's starting to look like it might be worth to store this information within the flash struct, in particular because after the rework a while ago we are guaranteed fixed sizes here so a lot of code could be pruned